### PR TITLE
Fix type piracy issue

### PR DESCRIPTION
**Pull request by tlnagy**
Fixes the issue brought up by @timholy in https://github.com/tlnagy/TiffImages.jl/pull/36#discussion_r583014076. Added this branch to testing as well.

### CODE REVIEW

File: src/files.jl [hunk 1]
```diff
@@ -86,7 +86,7 @@ function Base.read!(file::TiffFile, arr::AbstractArray)
     end
 end
 
-function Base.read!(io::IOStream, arr::SubArray{T,N,P,I,L}) where {T, N, P <: BitArray, I <: Tuple{UnitRange, Int64}, L}
+function Base.read!(file::TiffFile, arr::SubArray{T,N,P,I,L}) where {T, N, P <: BitArray, I , L}
```
Comment (timholy) [inline src/files.jl:89]:
```suggestion
function Base.read!(file::TiffFile, arr::SubArray{T,N,<:BitArray}) where {T, N}
```

**Review by timholy — APPROVED** [review]

**Comment by tlnagy** [comment]
This also fixes a pretty serious [bug](https://github.com/tlnagy/TiffImages.jl/pull/38/files#diff-2cf967cbb752d7b28a21a4faf064195a2d5ba4cfae4ed5d59d72e93f561304ceR170) that caused `TiffImages` to ignore stripped offsets, which actually yielded a substantial performance improvement. So this PR introduces a ~5x performance regression vs master. This condition is likely common enough that it's worth having a separate branch to handle the case were an image is striped but the stripes are contiguous and can therefore be read in as a single block.

**Comment by tlnagy** [comment]
Checking for contiguous strips allows me to recover the performance of master.

```julia
julia> using TiffImages

julia> @time TiffImages.load("/home/tlnagy/Downloads/Gcamp6F_BSA_60px_3_stack.ome.tif");
Progress: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| Time: 0:00:01
  2.724760 seconds (4.75 M allocations: 738.599 MiB, 2.18% gc time)

julia> @time TiffImages.load("/home/tlnagy/Downloads/Gcamp6F_BSA_60px_3_stack.ome.tif");
  0.768867 seconds (286.85 k allocations: 513.183 MiB, 3.43% gc time)

julia> using ImageMagick

julia> @time ImageMagick.load("/home/tlnagy/Downloads/Gcamp6F_BSA_60px_3_stack.ome.tif");
  4.143202 seconds (2.69 M allocations: 1.540 GiB, 5.18% gc time)

julia> @time ImageMagick.load("/home/tlnagy/Downloads/Gcamp6F_BSA_60px_3_stack.ome.tif");
  3.236719 seconds (6.72 k allocations: 1.408 GiB, 0.72% gc time)
```

**Comment by tlnagy** [comment]
I created an artificial image that has its stripes offset by a random amount that should catch any regressions in handling discontiguously striped uncompressed images.

**Pull request merged**